pkg/plugin/runtime_subprocess.go

```diff
@@ -97,20 +110,42 @@ func (r *RuntimeSubprocess) Dir() string {
 }
 
 // Invoke implementation for RuntimeConfig
-func (r *RuntimeSubprocess) Invoke(stdin io.Reader, stdout, stderr io.Writer, env []string) error {
-	// Prepare command based on runtime configuration
-	cmds := r.config.PlatformCommand
-	if len(cmds) == 0 && len(r.config.Command) > 0 {
-		cmds = []PlatformCommand{{Command: r.config.Command}}
-	}
+func (r *RuntimeSubprocess) Invoke(ctx context.Context, input *Input) (*Output, error) {
 
-	main, args, err := PrepareCommands(cmds, true, r.extraArgs)
+	pluginExec, err := convertInput(r, input)
 	if err != nil {
-		return fmt.Errorf("failed to prepare command: %w", err)
+		return nil, fmt.Errorf("failed to convert plugin input: %w", err)
+	}
+
+	pluginCommand := filepath.Join(r.plugin.Dir, pluginExec.command)
+	prog := exec.Command(
+		pluginCommand,
+		pluginExec.argv...)
+	prog.Env = pluginExec.env
+	buf := bytes.NewBuffer(nil)
+	prog.Stdout = buf
+	prog.Stderr = os.Stderr
+	if err := prog.Run(); err != nil {
+		if eerr, ok := err.(*exec.ExitError); ok {
+			os.Stderr.Write(eerr.Stderr)
+			return nil, fmt.Errorf("plugin %q exited with error", pluginCommand)
+		}
+		return nil, fmt.Errorf("failed to run plugin %q: %w", pluginCommand, err)
 	}
 
-	// Execute the command directly
-	return r.InvokeWithEnv(main, args, env, stdin, stdout, stderr)
+	return convertOutput(buf), nil
+
+	//main, args, err := PrepareCommands(cmds, true, r.extraArgs)
```

Comment:
How do we handle `extraArgs` if we're removing these `PrepareCommands()`? I don't think that is addressed in #3 or this PR?

> It merges extraArgs into any arguments supplied in the plugin. It
> returns the main command and an args array.